spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -601,6 +622,42 @@ DataFile[] files() {
     }
   }
 
+  public static class DeleteTaskCommit implements WriterCommitMessage {
+    private final DeleteFile[] taskFiles;
+    private final CharSequence[] referencedDataFiles;
+
+    DeleteTaskCommit(List<DeleteFile> deleteFiles, List<CharSequence> referencedDataFiles) {
+      this.taskFiles = deleteFiles.toArray(new DeleteFile[0]);
+      this.referencedDataFiles = referencedDataFiles.toArray(new CharSequence[0]);
+    }
+
+    // Reports bytesWritten and recordsWritten to the Spark output metrics.
+    // Can only be called in executor.
+    void reportOutputMetrics() {
+      long bytesWritten = 0L;
+      long recordsWritten = 0L;
+      for (DeleteFile dataFile : taskFiles) {
+        bytesWritten += dataFile.fileSizeInBytes();
+        recordsWritten += dataFile.recordCount();
+      }
+
+      TaskContext taskContext = TaskContext$.MODULE$.get();
+      if (taskContext != null) {
+        OutputMetrics outputMetrics = taskContext.taskMetrics().outputMetrics();
+        outputMetrics.setBytesWritten(bytesWritten);
+        outputMetrics.setRecordsWritten(recordsWritten);
+      }
+    }
+
+    DeleteFile[] files() {
+      return taskFiles;
+    }
+
+    CharSequence[] referencedDataFiles() {
```

Comment:
Is this being used? I don't this we need this to replace old files with new ones.

Reply:
Removed